src/main/java/core/basesyntax/service/impl/FruitFileReaderImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.exception.NoFileToReadException;
+import core.basesyntax.service.interfaces.FruitFileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class FruitFileReaderImpl implements FruitFileReader {
+    private CheckFilePathImpl filePathExist = new CheckFilePathImpl();
+
+    @Override
+    public List<String> readFile(String filePath) {
+        if (filePathExist.checkFilePath(filePath)) {
```

Comment:
Here, you're using your custom check for a string representing the path to a file and then throwing an exception if the file does not exist. It's okay, I think, but the static factory method `Path.of` already throws an `InvalidPathException` if the path string cannot be converted to a `Path`. So maybe we can simplify the logic by catching this exception and handling it. 
**For example, by adding some custom message to this error and rethrowing it further because it is unchecked.**
IMHO